### PR TITLE
FEATURE: add microdata prop and more links for crawler

### PR DESCRIPTION
**Pull request by fantasticfears**
Related: https://meta.discourse.org/t/breadcrumbs-microformat-for-seo/14634

### CODE REVIEW

**Comment by fantasticfears** [comment]
Also related: https://meta.discourse.org/t/semantic-microdata-and-more-content-for-crawlers/25211/1

File: lib/crawler_detection.rb [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+                  facebookexternalhit
+                  bingbot
+                  Baiduspider
+                  ia_archiver
+                  Wayback Save Page
+                  360Spider).join('|')
+
   def self.crawler?(user_agent)
-    !/Googlebot|Mediapartners|AdsBot|curl|Twitterbot|facebookexternalhit|bingbot|Baiduspider|ia_archiver|Wayback Save Page/.match(user_agent).nil?
+    !Regexp.new(CRAWLER_UA).match(user_agent).nil?
```
Comment (SamSaffron) [inline lib/crawler_detection.rb]: regex should be cached in a const.

File: config/routes.rb [hunk 2]
```diff
@@ -214,9 +211,10 @@
   post "login" => "static#enter"
   get "login" => "static#show", id: "login"
   get "faq" => "static#show", id: "faq"
-  get "guidelines" => "static#show", id: "guidelines"
-  get "tos" => "static#show", id: "tos"
-  get "privacy" => "static#show", id: "privacy"
+  get "guidelines" => "static#show", id: "guidelines", as: 'guidelines'
+  get "tos" => "static#show", id: "tos", as: 'tos'
+  get "privacy" => "static#show", id: "privacy", as: 'privacy'
+  get "about" => "static#about", as: 'about'
```
Comment (SamSaffron) [inline config/routes.rb]: not sure about moving about to static, it is representing very dynamic data

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -52,8 +52,19 @@
...
+            <%= link_to t('guidelines_topic.title'), guidelines_path %>
+            <%= link_to t('tos_topic.title'), tos_path %>
+            <%= link_to t('privacy_topic.title'), privacy_path %>
+            <%= link_to t('js.about.simple_title'), about_path %>
+          </nav>
+        </footer>
       </div>
-      <footer id='noscript-footer'>
+
+      <footer id='noscript-footer' class="container">
```
Comment (SamSaffron) [inline app/views/layouts/application.html.erb]: why does it need class container ?

File: app/controllers/list_controller.rb [hunk 4]
```diff
@@ -164,8 +164,8 @@ def parent_category_category_top
       user = list_target_user
       list = TopicQuery.new(user, top_options).list_top_for(period)
       list.for_period = period
-      list.more_topics_url = construct_next_url_with(top_options)
-      list.prev_topics_url = construct_prev_url_with(top_options)
+      list.more_topics_url = construct_url_with(:next, top_options)
+      list.prev_topics_url = construct_url_with(:prev, top_options)
```
Comment (SamSaffron) [inline app/controllers/list_controller.rb]: this is better than the construct_next_url_with construct_prev_url_with

**Comment by fantasticfears** [comment]
Rebased and updated according to sam's comments.

**Pull request merged**